Tasks/VsTest/VSTest.ts

```diff
@@ -0,0 +1,501 @@
+                    if(sourcesDirectory && !pathExistsAsFile(testAssembly)){
+                        var expandedPath = path.join(sourcesDirectory, testAssembly);
+                        if(pathExistsAsFile(expandedPath)){
+                            testAssemblyPath = expandedPath;
+                        }
+                    }
+                    vstest.arg(testAssemblyPath);  
+                });
+                
+                if(testFiltercriteria){
+                    vstest.arg("/TestCaseFilter:"+testFiltercriteria);                  
+                }
+                
+                if(parallelRunSettingsFile && pathExistsAsFile(parallelRunSettingsFile)){ 
+                    vstest.arg("/Settings:" + parallelRunSettingsFile);
+                }
+                
+                if(codeCoverageEnabled){
+                    vstest.arg("/EnableCodeCoverage");
+                }
+                
+                if(otherConsoleOptions){
+                    vstest.arg(otherConsoleOptions);
+                }
+                
+                vstest.arg("/logger:trx");
+                
+                if(pathtoCustomTestAdapters){
+                    if(pathExistsAsDirectory(pathtoCustomTestAdapters)){                    
+                        vstest.arg("/TestAdapterPath:\""+pathtoCustomTestAdapters+"\"");
+                    }
+                    else{
+                        vstest.arg("/TestAdapterPath:\""+path.dirname(pathtoCustomTestAdapters)+"\"");
+                    }
+                }
+                else if(sourcesDirectory && isNugetRestoredAdapterPresent(sourcesDirectory)){
+                    vstest.arg("/TestAdapterPath:\""+sourcesDirectory+"\"");
+                }
+                
+                tl.rmRF(testResultsDirectory, true);
+                tl.mkdirP(testResultsDirectory);
+                tl.cd(workingDirectory);
+                vstest.exec().then(function(code){
+                    cleanUp(parallelRunSettingsFile);
+                    defer.resolve(code);
+                })
+                .fail(function(err) {
+                    cleanUp(parallelRunSettingsFile);
+                    tl.warning("Vstest failed with error. Check logs for failures. There might be failed tests");
+                    tl.error(err);
+                    defer.resolve(1);
+                });
+            });  
+        });    
+    });
+        
+    return defer.promise;
+}
+
+function getFilteredFiles(filesFilter: string, allFiles: string[]): string[] {
+    if(os.type().match(/^Win/)){
+        return tl.match(allFiles, filesFilter, {matchBase: true, nocase: true});
+    }
+    else{
+        return tl.match(allFiles, filesFilter, {matchBase: true});
+    }  
+}
+
+// removes elements of array2 from array1
+function removeArrayFromArray(array1: string[], array2: string[]): string[]{
```

Comment:
you don't need to write old school JS. You can use ES6 functions

let diff = array1.filter(x => array2.indexOf(x) == -1);